### ISSUE TITLE
Create toy example using OceanBioME.jl 0D wrapper

### ISSUE THREAD

**Issue by nanophyto** [agate-model/Agate.jl#24]
We need an initial toy 0D model. We can rely on an existing wrapper by OceanBioME.jl initially (which relies on Oceananigans.jl). This can then be used for tests with Turing.jl and AD packages.

There is an initial version of the toy model here:
https://github.com/agate-model/Agate.jl/blob/dynamic_parsing/src/box_model/box_wrapper.jl

Many things to tidy:

`base_maximum_growth=0.69, maximum_grazing_rate= 2.1522`  should be an array
`PAR` definition should be outside of wrapper
` N = 7.0, P = 0.01, Z = 0.05` these initial conditions should be passed as an array into the wrapper
`if timeseries` only exports P currently

**Comment by radka-j** [comment]
I noticed that the OceanBioME [BoxModel example](https://oceanbiome.github.io/OceanBioME.jl/stable/generated/box/) sets up PAR as a function field (on a 1x1x1 grid) whereas in the data assimilation example it is passed as a forcing function and I was wondering what the difference between the two is. While looking into it I noticed a strange thing in how BoxModel handles the forcing functions passed to it that I think might be a bug (or the meaning of forcing does not include PAR).

Note that the models assign PAR as an auxiliary field, e.g., for the NPZD model we have:
 
```Julia
required_biogeochemical_tracers(biogeochemistry)=(:N, :P, :Z, :D, :T)
required_biogeochemical_auxiliary_fields(biogeochemistry) = (:PAR,)
``` 
In the data assimilation example we define this function

 ```Julia
PAR(t) = PAR⁰(t) * exp(0.2z)
```
and pass it to the BoxModel as 
```Julia
model = BoxModel(... forcing=(;PAR)) # equivalent to forcing=(PAR=PAR)
``` 
i.e., `forcing` is a NamedTuple where `forcing[:PAR]` returns our function.


Inside the [BoxModel function](https://github.com/OceanBioME/OceanBioME.jl/blob/3c287d625aa7135c8297fa36b94347979344c235/src/BoxModel/boxmodel.jl#L62) this happens:

```Julia 
@inline no_func(args...) = 0.0

# NOTE variables here are tracers NOT auxiliary fields
variables = required_biogeochemical_tracers(biogeochemistry)
forcing = NamedTuple{variables}([variable in keys(forcing) ? forcing[variable] : no_func for variable in variables])
```
which sets forcing to this tuple
```Julia
model.forcing = (N = no_func, P = no_func, Z = no_func, D = no_func, T = no_func)
```

In other words, it looks like the PAR function get entirely ignored if it is passed in this way.

**Comment by radka-j** [comment]
I found the[ commit](https://github.com/OceanBioME/OceanBioME.jl/commit/116fa193c2d0cf24bdebcf8290399c175d49dcc2) that resulted in the above behaviour. Previously `variables` included both tracers and auxiliary variables as one would expect 
```Julia
variables = (required_biogeochemical_tracers(biogeochemistry)..., required_biogeochemical_auxiliary_fields(biogeochemistry)...)
```

This was changed when the `PrescribedPhotosyntheticallyActiveRadiation` was added and the BoxModel example was updated from passing PAR as a `forcing` function to the version that is now in the docs. So I will follow that.

**Comment by jagoosw** [comment]
Hey @radka-j,

Yeah sorry for this oversight in the data assimilation example. No one had really used the box model until this summer so it was a mess but I've hopefully tidied it up now and made the API more obvious (it was confusing "forcing" the PAR etc.).

I just merged the data assimilation example and it should work now

**Comment by radka-j** [comment]
Thank you @jagoosw for all your hard work on OceanBioME, it's a great package and a really useful resource!
